packages/react/src/transactions/infrastructure/createPublicationMetadata.ts

```diff
@@ -94,42 +101,54 @@ function contentFields(request: CreatePublicationRequest) {
   }
 }
 
-function resolveNonCollectableMetadata(
-  request: NonCollectablePublicationRequest,
-): PublicationMetadata {
+function resolveNonCollectableMetadata() {
   return {
-    ...essentialFields(request),
-    ...optionalFields(request),
-    ...contentFields(request),
     name: 'none', // although "name" is not needed when a publication is not collectable, Publication Metadata V2 schema requires it ¯\_(ツ)_/¯
-    attributes: [],
-  };
+  } as const;
 }
 
-function resolveCollectableMetadata(request: CollectablePublicationRequest): PublicationMetadata {
+function resolveDeprecatedCollectableMetadata(request: CollectablePublicationRequest) {
+  return {
+    attributes: mapMetadataAttributes(request.collect.metadata.attributes ?? []),
+    image: request.collect.metadata.image,
+    imageMimeType: request.collect.metadata.imageMimeType,
+  } as const;
+}
+
+function resolveCollectableMetadata(request: CollectablePublicationRequest) {
+  if ('attributes' in request.collect.metadata && 'attributes' in request) {
+    console.warn(
+      'Both "attributes" and "collect.metadata.attributes" are provided. Using "collect.metadata.attributes"',
```

Comment:
I ended up using existing format over new one as it felt the most backwards compatible way. Early in the SDK development I am aware of some JS integration that did pass these at the top level for lack of better documentation at the time.

With no visibility on those I prefer to avoid breaking examples, demos, docs until we release 2.0.